src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp

```diff
@@ -640,4 +641,111 @@ TEST_CASE("RotatedSPOs construct delta matrix", "[wavefunction]")
     CHECK(new_params2[i] == Approx(old_params[i]));
 }
 
+namespace testing
+{
+opt_variables_type& getMyVars(SPOSet& rot) { return rot.myVars; }
+opt_variables_type& getMyVarsFull(RotatedSPOs& rot) { return rot.myVarsFull; }
+std::vector<std::vector<QMCTraits::RealType>>& getHistoryParams(RotatedSPOs& rot) { return rot.history_params_; }
+} // namespace testing
+
+// Test using global rotation
+TEST_CASE("RotatedSPOs read and write parameters", "[wavefunction]")
+{
+  auto fake_spo = std::make_unique<FakeSPO>();
+  fake_spo->setOrbitalSetSize(4);
+  RotatedSPOs rot("fake_rot", std::move(fake_spo));
+  int nel = 2;
+  rot.buildOptVariables(nel);
+
+  optimize::VariableSet vs;
+  rot.checkInVariablesExclusive(vs);
+  vs[0] = 0.1;
+  vs[1] = 0.15;
+  vs[2] = 0.2;
+  vs[3] = 0.25;
+  rot.resetParametersExclusive(vs);
+
+  hdf_archive hout;
+  vs.writeToHDF("rot_vp.h5", hout);
+
+  rot.writeVariationalParameters(hout);
+
+  hout.close();
+
+  auto fake_spo2 = std::make_unique<FakeSPO>();
+  fake_spo2->setOrbitalSetSize(4);
+
+  RotatedSPOs rot2("fake_rot", std::move(fake_spo2));
+  rot2.buildOptVariables(nel);
+
+  optimize::VariableSet vs2;
+  rot2.checkInVariablesExclusive(vs2);
+
+  hdf_archive hin;
```

Comment:
Also use a scope around hin?

Reply:
Closing hout is necessary for functionality (the test returns an error inside the HDF code if it is not closed)
Closing hin is not necessary for that, at least.  If it has a scope, the end of the scope around hin will end at the end of the function, so it seems a bit redundant.

Reply:
Agreed. It is more of a preference.